Step2 학습 테스트 PR 요청드립니다.

zerodkdj opened the pull request:
안녕하세요 이전에 주신 피드백 확인해서 반영해보았습니다. 감사합니다 :)

zerodkdj committed:
docs: README.md파일작성

- 기능요구사항정리

zerodkdj committed:
test: splitAndSum메소드 null값,빈값 테스트코드 작성

zerodkdj committed:
test: splitAndSum메소드에서 구분자로 구분된 숫자합 테스트 작성

zerodkdj committed:
refactor: splitAndSum테스트메소드 DisplayName수정

- 구체적인 방식에서 일반화를 위해 추상적으로 변경

zerodkdj committed:
refactor: splitAndSum테스트메소드 ParameterizedTest로 변경

zerodkdj committed:
test: splitAndSum메소드 테스트추가

- 커스텀 구분자를 입력한 경우 테스트
- 구분된 숫자중 음수포함인 경우 예외발생 테스트

zerodkdj committed:
test: splitAndSum메소드 숫자 이외에 값 포함된 경우 테스트

zerodkdj committed:
test: splitAndSum메소드 숫자만 입력할 경우 테스트

zerodkdj committed:
refactor: StringAddCalculator Exception메시지추가

- convertToInt에서 음수포함될 때 RuntimeException에 메시지를 추가

zerodkdj committed:
refactor: 테스트메소드앞에 test_that추가

- 테스트 메소드가 하는 일을 명확하게 하기위해 문구추가

seokhongkim commented on README.md:
```diff
@@ -6,4 +6,11 @@
\ No newline at end of file
+* [텍스트와 이미지로 살펴보는 온라인 코드 리뷰 과정](https://github.com/next-step/nextstep-docs/tree/master/codereview)
+
+### 2단계 문자열 덧셈 계산기 기능정리
```
기능 목록 잘 작성해주셨습니다. 👍 
마크다운의 체크리스트 문법을 사용해보시면 완료된 기능을 조금 더 쉽게 체크하면서 진행하실 수 있을 것 같아요! :)

seokhongkim commented on src/main/java/step2/StringAddCalculator.java:
```diff
@@ -0,0 +1,45 @@
+
+    private static final int DEFAULT_VALUE = 0;
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final Pattern CUSTOM_PATTERN = Pattern.compile("//(.)\\n(.*)");
```
패턴 미리 만들어서 사용하시는 것 좋습니다. 👍

seokhongkim commented on src/main/java/step2/StringAddCalculator.java:
```diff
@@ -0,0 +1,45 @@
+public class StringAddCalculator {
+
+    private static final int DEFAULT_VALUE = 0;
+    private static final String DEFAULT_DELIMITER = ",|:";
```
적절한 상수 사용 좋습니다. 👍

seokhongkim commented on src/test/java/step2/StringAddCalculatorTest.java:
```diff
@@ -0,0 +1,73 @@
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+public class StringAddCalculatorTest {
```
테스트 잘 작성해주셨고, `DisplayName` 과 테스트 메소드명 모두 이해하기 쉽게 작성해주셨습니다. 👍

seokhongkim commented on src/main/java/step2/StringAddCalculator.java:
```diff
@@ -0,0 +1,45 @@
+        return sumNumbers(tokens);
+    }
+
+    private static boolean isEmptyOrNull(String text) {
```
꼼꼼한 밸리데이션 좋습니다. 👍

seokhongkim commented on src/main/java/step2/StringAddCalculator.java:
```diff
@@ -0,0 +1,45 @@
+    private static final String NEGATIVE_NUMBER_MESSAGE = "음수는 포함될 수 없습니다";
+
+    public static int splitAndSum(String text) {
+        if (isEmptyOrNull(text)) return DEFAULT_VALUE;
```
if 절에 대한 컨벤션이 다양하게 사용되고 있네요.
if 키워드와 `(` 사이에 공백이 있는 경우와 없는 경우가 있고, 이후 중괄호로 코드를 명시적으로 블록으로 감싼 경우와 그렇지 않은 경우가 있네요.
컨벤션을 설정하고 통일해보시면 좋겠습니다. :)

seokhongkim commented on src/main/java/step2/StringAddCalculator.java:
```diff
@@ -0,0 +1,45 @@
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class StringAddCalculator {
```
클래스가 두가지 역할을 하고 있는 것 같아요.
1. 입력받은 문자열을 파싱한다. (숫자로 변환 포함)
2. 숫자를 더한 값을 계산한다.

각 역할을 클래스로 나눠보시면 어떨까요?

seokhongkim reviewed (changes requested):
동기님, 2단계 미션 잘 구현해주셨습니다. 👍 

코멘트 드릴 부분이 많지는 않지만 한 번 고민해보시면 좋을 것 같아서 반영 요청 드립니다. :)
코멘트 확인 부탁드립니다. 🙏 

코멘트 보시고 궁금하신 점은 깃헙 코멘트나 슬랙 DM으로 편하게 말씀 부탁드립니다. 😀

zerodkdj committed:
refactor: 코드 컨벤션 통일

- if문 다음에 공백없이 괄호 붙이기
- if문 다음에 한줄짜리도 중괄호 붙이기

zerodkdj committed:
refactor: StringParser,Sum클래스추가

- 문자열을 파싱하는 역할을 하는 StringParser클래스가 문자열을 숫자배열로 파싱하는 역할 담담
- Sum클래스에서 파싱된 숫자배열의 값 더하기 담당

zerodkdj committed:
refactor: from메소드 스트림내부에 들어가는 람다식 메소드참조로 다른 클래스와 통일

zerodkdj committed:
refactor: 외부에서 사용하지 않는 parseToStringArray메소드 접근지정자 private으로 변경

seokhongkim commented on src/main/java/step2/Sum.java:
```diff
@@ -0,0 +1,21 @@
+                .sum();
+    }
+
+    private static boolean filterPositive(int number){
```
다음 단계에서는 이런 도메인 규칙을 값 객체(e.g., `PositiveNumber`)를 추가하고 거기에서 관리하시는 방법으로 접근해보셔도 좋겠습니다. :)

seokhongkim commented on src/main/java/step2/StringParser.java:
```diff
@@ -0,0 +1,26 @@
+    private static final String DEFAULT_DELIMITER = ",|:";
+    private static final Pattern CUSTOM_PATTERN = Pattern.compile("//(.)\\n(.*)");
+
+    public static int[] parseToIntArray(String text){
```
메소드 구현 코드에 `)` 와 `{` 사이에 공백이 없네요.
IDE를 사용하신다면 코드 스타일 설정 후, 코드를 수정하실 때마다 줄 맞춤을 한 번씩 진행해보시면 더 도움이 될 것 같습니다. :)

seokhongkim reviewed (approved):
동기님, 2단계 미션 피드백 잘 반영해주셨습니다. 💯 
2단계 PR은 이만 머지하도록 하겠습니다.

코멘트 조금 남겼는데, 이 부분은 3단계 미션 진행하실 때 참고 부탁드립니다.
3단계 미션에서 뵙겠습니다. 😀
